components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1292,10 +1342,16 @@ bool KeyringService::IsHardwareAccount(const std::string& account) const {
 void KeyringService::Unlock(const std::string& password,
                             KeyringService::UnlockCallback callback) {
   if (!ResumeKeyring(mojom::kDefaultKeyringId, password)) {
-    encryptor_.reset();
+    encryptors_.erase(mojom::kDefaultKeyringId);
     std::move(callback).Run(false);
     return;
   }
+  if (IsFilecoinEnabled()) {
+    if (!ResumeKeyring(mojom::kFilecoinKeyringId, password)) {
+      VLOG(1) << __func__ << " Unable to unlock filecoin keyring";
```

Comment:
when unlocking failed, we should erase filecoin's encryptor

Reply:
done

Reply:
done